nixos/modules/services/web-apps/node-red.nix

```diff
@@ -0,0 +1,112 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.node-red;
+  defaultUser = "node-red";
+in
+{
+  options.services.node-red = {
+    enable = mkEnableOption "the Node-RED service";
+
+
+    package = mkOption {
+      default = pkgs.nodePackages.node-red;
+      defaultText = "pkgs.nodePackages.node-red";
+      type = types.package;
+      description = "Node-RED package to use.";
+    };
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Open ports in the firewall for the server.
+      '';
+    };
+
+    configFile = mkOption {
+      type = types.path;
+      default = "${cfg.package}/lib/node_modules/node-red/settings.js";
+      defaultText = "\${cfg.package}/lib/node_modules/node-red/settings.js";
+      description = ''
+        Path to the JavaScript configuration file.
+        See <link
+        xlink:href="https://github.com/node-red/node-red/blob/master/packages/node_modules/node-red/settings.js"/>
+        for a configuration example.
+      '';
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 1880;
+      description = "Listening port.";
+    };
+
+    user = mkOption {
+      type = types.str;
+      default = defaultUser;
+      description = ''
+        Node-RED will be run under this user (user will be created if it doesn't exist.
+        This can be your user name).
```

Comment:
Should we make it such that the user *is* created for you? Or is it correct to leave this to the system administrator?

Reply:
Typical NixOS options would say something like `User under which Node-RED runs.` and you should probably follow that convention. I'll additionally note that I do not like the "This can be your user name" comment as this leads to the bad practice of users running system services as non system users.

I think all the NixOS `user` and `group` options should be updated to say something like:
```
User under which ____ runs. If left as the default value this user will automatically be created
on system activation, otherwise the sysadmin is responsible for ensuring the user exists.
```
... but that is probably best left for a separate PR :smile:

Reply:
Has the latest commit addressed your feedback? 7801b79

Reply:
LGTM :+1: